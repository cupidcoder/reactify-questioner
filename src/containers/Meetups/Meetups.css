.Meetups {
  min-height: 70vh;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 90%;
  margin: 0 auto;
}
