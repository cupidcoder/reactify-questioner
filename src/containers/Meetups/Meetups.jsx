import React, { Component } from 'react';
import { connect } from 'react-redux';
import Skeleton from 'react-skeleton-loader';
import BaseContainer from '../BaseContainer/BaseContainer';
import MeetupCard from '../../components/MeetupCard/MeetupCard';
import { processGetMeetups } from '../../actions/getMeetupActions';
import style from './Meetups.css';

class Meetups extends Component {
  async componentDidMount() {
    const { getMeetupsRequest } = this.props;
    await getMeetupsRequest();
  }

  render() {
    const { isLoading, data } = this.props;
    let meetupCards;
    if (isLoading) {
      meetupCards = (
        <Skeleton
          count={1}
          width='50%'
          height='4rem'
          widthRandomness={0}
          borderRadius={0}
        />
      );
    } else if (data && data.length === 0) {
      meetupCards = <h3>No meetups available at the moment</h3>;
    } else {
      meetupCards = data.map(meetup => (
        <MeetupCard
          key={meetup.id}
          id={meetup.id}
          name={meetup.topic}
          description={meetup.description}
          location={meetup.location}
        />
      ));
    }
    return (
      <BaseContainer>
        <div className={style.Meetups}>
          {meetupCards}
        </div>
      </BaseContainer>
    );
  }
}

const mapStateToProps = state => ({
  isLoading: state.getMeetups.isLoading,
  status: state.getMeetups.status,
  message: state.getMeetups.message,
  data: state.getMeetups.data
});

const mapDispatchToProps = dispatch => ({
  getMeetupsRequest: () => dispatch(processGetMeetups())
});

const MeetupsComponent = connect(mapStateToProps, mapDispatchToProps)(Meetups);

export { Meetups, MeetupsComponent };
