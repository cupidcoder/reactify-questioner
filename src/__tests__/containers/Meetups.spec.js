import React from 'react';
import { shallow } from 'enzyme';
import { Meetups } from '../../containers/Meetups/Meetups';

describe('Meetups', () => {
  it('Should correctly save snapshot of the Meetups page', () => {
    const props = {
      isLoading: false,
      data: []
    };
    const wrapper = shallow(<Meetups {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
