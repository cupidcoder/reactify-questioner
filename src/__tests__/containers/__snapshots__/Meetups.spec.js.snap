// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meetups Should correctly save snapshot of the Meetups page 1`] = `ShallowWrapper {}`;
